fix NiftyReg with OpenMP

using
```sh
cmake \
  -DDEVEL_BUILD=ON \
  -DCMAKE_BUILD_TYPE=Release -DBUILD_STIR_WITH_OPENMP=ON -DUSE_SYSTEM_ACE=ON \
  -DUSE_SYSTEM_Armadillo=ON -DUSE_SYSTEM_Boost=ON -DUSE_SYSTEM_FFTW3=ON \
  -DUSE_SYSTEM_HDF5=ON -DUSE_ITK=ON -DBUILD_siemens_to_ismrmrd=ON \
  -DUSE_SYSTEM_SWIG=ON \
  -USIRF_URL -DSIRF_URL=/devel/sirf/.git \
  -USIRF_TAG -DSIRF_TAG=setup_py-configure_and_install \
  .
```

On Ubuntu 18.04, the output is:

```
_reg_maths_eigen.cpp:(.text._Z3svdIfEvPPT_mmPS2_S3_S3_[_Z3svdIfEvPPT_mmPS2_S3_S3_]+0x5e9): undefined reference to `GOMP_parallel'
/opt/SIRF-SuperBuild/INSTALL/lib/lib_reg_maths.a(_reg_maths_eigen.cpp.o):_reg_maths_eigen.cpp:(.text._Z3svdIfEvPPT_mmPS2_S3_S3_[_Z3svdIfEvPPT_mmPS2_S3_S3_]+0x610): more und
efined references to `GOMP_parallel' follow
collect2: error: ld returned 1 exit status
src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/build.make:126: recipe for target 'src/Registration/cReg/sirf_tensor_split_join' failed
make[5]: *** [src/Registration/cReg/sirf_tensor_split_join] Error 1
CMakeFiles/Makefile2:758: recipe for target 'src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/all' failed
make[4]: *** [src/Registration/cReg/CMakeFiles/sirf_tensor_split_join.dir/all] Error 2
make[4]: *** Waiting for unfinished jobs....
[ 67%] Building CXX object src/xSTIR/pSTIR/CMakeFiles/_pystir.dir/opt/SIRF-SuperBuild/sources/STIR/src/modelling_buildblock/modelling_registries.cxx.o
[ 68%] Building CXX object src/xSTIR/pSTIR/CMakeFiles/_pystir.dir/opt/SIRF-SuperBuild/sources/STIR/src/spatial_transformation_buildblock/spatial_transformation_registries.c
xx.o
[ 69%] Linking CXX shared module _pystir.so
[ 69%] Built target _pystir
make[3]: *** [all] Error 2
Makefile:138: recipe for target 'all' failed
CMakeFiles/SIRF.dir/build.make:118: recipe for target 'builds/SIRF/stamp/SIRF-build' failed
make[2]: *** [builds/SIRF/stamp/SIRF-build] Error 2
make[1]: *** [CMakeFiles/SIRF.dir/all] Error 2
CMakeFiles/Makefile2:306: recipe for target 'CMakeFiles/SIRF.dir/all' failed
Makefile:94: recipe for target 'all' failed
make: *** [all] Error 2
```

(from #195)

--- a/SuperBuild/External_NiftyReg.cmake
+++ b/SuperBuild/External_NiftyReg.cmake
@@ -58,17 +58,15 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
     DOWNLOAD_DIR ${${proj}_DOWNLOAD_DIR}
     STAMP_DIR ${${proj}_STAMP_DIR}
     TMP_DIR ${${proj}_TMP_DIR}
-
     CMAKE_ARGS
-        -DCMAKE_PREFIX_PATH=${SUPERBUILD_INSTALL_DIR}
-        -DCMAKE_LIBRARY_PATH=${SUPERBUILD_INSTALL_DIR}/lib
-        -DCMAKE_INCLUDE_PATH=${SUPERBUILD_INSTALL_DIR}
-        -DCMAKE_INSTALL_PREFIX=${${proj}_Install_Dir}
-        -DUSE_THROW_EXCEP=ON
-	    INSTALL_DIR ${${proj}_Install_Dir}
-    DEPENDS
-        ${${proj}_DEPENDENCIES}
-  )
+      -DCMAKE_PREFIX_PATH=${SUPERBUILD_INSTALL_DIR}
+      -DCMAKE_LIBRARY_PATH=${SUPERBUILD_INSTALL_DIR}/lib
+      -DCMAKE_INCLUDE_PATH=${SUPERBUILD_INSTALL_DIR}
+      -DCMAKE_INSTALL_PREFIX=${${proj}_Install_Dir}
+      -DUSE_THROW_EXCEP=ON
+      -DUSE_OPENMP:BOOL=OFF
+    INSTALL_DIR ${${proj}_Install_Dir}
+    DEPENDS ${${proj}_DEPENDENCIES})
 
     set(${proj}_ROOT        ${${proj}_SOURCE_DIR})
     set(${proj}_INCLUDE_DIR ${${proj}_SOURCE_DIR})

--- a/SuperBuild/External_NiftyReg.cmake
+++ b/SuperBuild/External_NiftyReg.cmake
@@ -64,6 +64,7 @@ if(NOT ( DEFINED "USE_SYSTEM_${externalProjName}" AND "${USE_SYSTEM_${externalPr
       -DCMAKE_INCLUDE_PATH=${SUPERBUILD_INSTALL_DIR}
       -DCMAKE_INSTALL_PREFIX=${${proj}_Install_Dir}
       -DUSE_THROW_EXCEP=ON
+      # fixes lib_reg_maths.a `GOMP_parallel' undefined reference linker errors
       -DUSE_OPENMP:BOOL=OFF
     INSTALL_DIR ${${proj}_Install_Dir}
     DEPENDS ${${proj}_DEPENDENCIES})

